kale/embed/mpca.py

```diff
@@ -178,52 +181,65 @@ def _fit(self, X):
 
                 eig_vals, eig_vecs = np.linalg.eig(Phi[i])
                 idx_sorted = eig_vals.argsort()[::-1]
-                lmds[i] = eig_vals[idx_sorted]
+                lambdas[i] = eig_vals[idx_sorted]
                 proj_mats[i] = eig_vecs[:, idx_sorted]
                 proj_mats[i] = (proj_mats[i][:, :shape_out[i]]).T
 
         x_transformed = multi_mode_dot(X, proj_mats, modes=[m for m in range(self.ndim - 1)])
-        x_trans_vecs = unfold(x_transformed, mode=-1)
-        x_trans_vars = np.diag(np.dot(x_trans_vecs, x_trans_vecs.T))
-        idx_order = x_trans_vars.argsort()[::-1]
+        x_trans_vec = unfold(x_transformed, mode=-1)  # vectorise the transformed features
+        # diagonal of the covariance of vectorised features
+        x_trans_cov_diag = np.diag(np.dot(x_trans_vec, x_trans_vec.T))
+        idx_order = x_trans_cov_diag.argsort()[::-1]
 
         self.proj_mats = proj_mats
         self.idx_order = idx_order
         self.shape_out = shape_out
 
         return self
 
-    def transform(self, X):
+    def transform(self, X, vectorise=False, n_components=None):
         """Perform dimension reduction on X
 
         Args:
-            X (ndarray): shape (P_1, P_2, ..., P_N, n_samples)
+            X (ndarray): Data to perform dimension reduction, shape (I_1, I_2, ..., I_N, n_samples).
+            vectorise (bool): Whether vectorise the transformed tensor. Defaults to Flase.
+            n_components (int): Number of components to keep. Applies only when vectorise=True. 
+                Defaults to None.
 
         Returns:
-            ndarray: Transformed data, shape (i_1, i_2, ..., i_N, n_samples)
+            ndarray: Transformed data, shape (P_1, P_2, ..., P_N, n_samples).
```

Comment:
That's if `vectorise=false`.
If `vectorise=true`, the shape should be (`n_components`, `n_samples`), right?

Reply:
should be (n_samples, P_1 * P_2 * ... * P_N) or (n_samples, n_components), which is consistent with scikit-learn pca, will update the docstring soon